tests/tests_interlocking/test_interlocking_configuration.py

```diff
@@ -1,8 +1,55 @@
-from src.interlocking_component.route_controller import InterlockingConfiguration
+from uuid import UUID
 
+import pytest
 
-def test_default_true():
-    """This test if an InterlockingConfiguration Object can be initialized
-    and if the default for dynamicRouting is false."""
-    interlocking_configuration = InterlockingConfiguration()
-    assert not interlocking_configuration.dynamicRouting
+from src.interlocking_component.interlocking_configuration import (
+    InterlockingConfiguration,
+)
+from tests.decorators import recreate_db_setup
+
+
+@pytest.mark.parametrize(
+    "object_as_dict",
+    [
+        {
+            "dynamicRouting": True,
+        },
+    ],
+)
+class TestCorrectFilledDict:
+    """Test that a object of a class can be created and deserialized with valid data."""
+
+    @recreate_db_setup
+    def setup_method(self):
+        pass
+
+    def test_create(self, object_as_dict: dict):
+        """Test that a object of a class can be created."""
+        obj = InterlockingConfiguration.create(
+            **object_as_dict,
+        )
+        assert (
+            InterlockingConfiguration.select()
+            .where(InterlockingConfiguration.id == obj.id)
+            .first()
+            == obj
+        )
+
+    def test_serialization(self, object_as_dict: dict):
+        """Test that an object of a class can be serialized."""
+        obj = InterlockingConfiguration.create(
```

Comment:
You are not serializing here.

You create an object using the peewee constructor. Then you check whether the resulting `InterlockingConfiguration` object has the same content as the dict. 

What you want to test instead is whether the conversion of the object to a dict works. Therefore you could call `obj.to_dict`. Then you could test if the resulting dict has the same content as the original dict.

Reply:
Thank you for the feedback, I still don't understand peewee+marshmallow 100%, but I have changed it now. @instant-sky, this comment and the one below also applies to your tests. I think you should change this too.